local_vars/local_vars.cc

```diff
@@ -226,35 +232,246 @@ class LocalNameInserter {
+                    original.loc, core::make_type<core::LiteralType>(core::Symbols::Symbol(), arg.arg._name)));
+                kwArgValueEntries.emplace_back(ast::make_expression<ast::Local>(original.loc, arg.arg));
+            } else if (arg.flags.isKeywordSplat()) {
+                ENFORCE(!kwArgsHash, "Saw multiple keyword splats");
+
+                // TODO(aprocter): is it necessary to duplicate the hash here?
+                kwArgsHash =
+                    ast::MK::Send1(original.loc, ast::MK::Constant(original.loc, core::Symbols::Magic()),
+                                   core::Names::toHashDup(), ast::make_expression<ast::Local>(original.loc, arg.arg));
+                if (!kwArgKeyEntries.empty()) {
+                    // TODO(aprocter): it might make more sense to replace this with an InsSeq that calls
+                    // <Magic>::<merge-hash>, which is what's done in the desugarer.
+                    kwArgsHash = ast::MK::Send1(
+                        original.loc,
+                        ast::MK::Hash(original.loc, std::move(kwArgKeyEntries), std::move(kwArgValueEntries)),
+                        core::Names::merge(), std::move(kwArgsHash));
+                    kwArgKeyEntries.clear();
+                    kwArgValueEntries.clear();
+                }
+            } else if (arg.flags.block) {
+                blockArg = ast::make_expression<ast::Local>(original.loc, arg.arg);
+            } else if (arg.flags.shadow) {
+                ENFORCE(false, "Shadow only come from blocks, but super only looks at a method's args");
             } else {
-                if (auto e = ctx.beginError(original.loc, core::errors::Namer::SelfOutsideClass)) {
-                    e.setHeader("`{}` outside of method", "super");
+                ENFORCE(false, "Unhandled arg kind in ZSuperArgs");
+            }
+        }
+
+        // At this stage the method should always have a block arg, even if it's synthetic (i.e., wasn't mentioned in
+        // the original source).
+        ENFORCE(blockArg, "Block argument not present");
+
+        // If there were any posargs after a positional splat, fold them into the splatted array.
+        if (posArgsArray && !posArgsEntries.empty()) {
+            posArgsArray = ast::MK::Send1(original.loc, std::move(posArgsArray), core::Names::concat(),
+                                          ast::MK::Array(original.loc, std::move(posArgsEntries)));
+            posArgsEntries.clear();
+        }
+
+        // If there were any keyword args after a keyword splat, fold them into the splatted hash.
+        if (kwArgsHash && !kwArgKeyEntries.empty()) {
+            kwArgsHash =
+                ast::MK::Send1(original.loc, std::move(kwArgsHash), core::Names::merge(),
+                               ast::MK::Hash(original.loc, std::move(kwArgKeyEntries), std::move(kwArgValueEntries)));
+            kwArgKeyEntries.clear();
+            kwArgValueEntries.clear();
+        }
```

Comment:
Thinking about this a bit more, this whole "if" is actually a can't-happen, and we even `ENFORCE` as much as we're walking the args. Will delete this.